File: UIHotel/Data/Table/Guest.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace UIHotel.Data.Table
{
    [Table("guest")]
    public class Guest
    {
        [Key]
        [Column("id", Order = 0)]
        public long Id { get; set; }

        [Required]
        [Column("id_number", Order = 1)]
        [StringLength(50)]
        public string IdNumber { get; set; }

        [Required]
        [Column("id_kind", Order = 2)]
        [StringLength(20)]
        public string IdKind { get; set; } = "KTP";

        [Required]
        [Column("fullname", Order = 3)]
        [StringLength(50)]
        public string Fullname { get; set; }

        [Column("email", Order = 4)]
        [StringLength(100)]
        public string Email { get; set; }

        [Column("isVIP", Order = 5)]
        public bool IsVIP { get; set; }

        [Column("birth_place", Order = 6)]
        [StringLength(50)]
        public string BirthPlace { get; set; }

        [Required]
        [DataType(DataType.Date)]
        [Column("birth_day", Order = 7, TypeName = "Date")]
        public DateTime BirthDay { get; set; }

        [Required]
        [Column("phone1", Order = 8)]
        [StringLength(15)]
        public string Phone1 { get; set; }

        [Column("phone2", Order = 9)]
        [StringLength(15)]
        public string Phone2 { get; set; }

        [Column("address", Order = 10)]
        [StringLength(255)]
        public string Address { get; set; }

        [Column("city", Order = 11)]
        [StringLength(50)]
        public string City { get; set; }

        [Column("province", Order = 12)]
        [StringLength(50)]
        public string Province { get; set; }

        [Column("state", Order = 13)]
        [StringLength(50)]
        public string State { get; set; }

        [Column("postcode", Order = 14)]
        [StringLength(10)]
        public string PostCode { get; set; }

        [Required]
        [Column("photo_doc", Order = 15)]
        [StringLength(50)]
        public string PhotoDoc { get; set; }

        [Column("photo_guest", Order = 16)]
        [StringLength(50)]
        public string PhotoGuest { get; set; }

        [Required]
        [Column("create_at", Order = 17)]
        public DateTime CreateAt { get; set; }

        [Column("update_at", Order = 18)]
        public DateTime? UpdateAt { get; set; }

        [NotMapped]
        public string FullAddress
        {
            get
            {
                return this.Address + ", " + this.City + ", " + this.Province + ", " + this.State;
            }
        }

        [NotMapped]
        public string DetailLink
        {
            get
            {
                return "http://localhost.com/guest/get/detail?id=" + this.Id;
            }
        }

        [NotMapped]
        public string EditLink
        {
            get
            {
                return "http://localhost.com/guest/get/edit?id=" + this.Id;
            }
        }

        [NotMapped]
        public string PhotoUrl
        {
            get
            {
                return "http://localhost.com/Upload/" + this.PhotoGuest;
            }
        }

        [NotMapped]
        public string DocumentUrl
        {
            get
            {
                return "http://localhost.com/Upload/" + this.PhotoDoc;
            }
        }
    }
}

File: UIHotel2/Assets/vue/src/lib/Model/Guest.ts
import { ss, si, execute, executeScalar } from "@/lib/Test";
import moment from "moment";

export interface IGuest {
  IdNumber: string;
  Fullname: string;
  Email: string;
  IsVIP: boolean;
  BirthPlace: string;
  BirthDay: string;
  Phone1: string;
  Phone2: string;
  Address: string;
  City: string;
  Province: string;
  State: string;
  PhotoDoc: string;
  PhotoGuest: string;
}

export class Guest {
  public static Store(user: IGuest): number {
    var new_moment = moment();
    var qry = si()
      .into("guests")
      .set("IdKind", "KTP")
      .set("IdNumber", user.IdNumber)
      .set("Fullname", user.Fullname)
      .set("Email", user.Email)
      .set("IsVIP", user.IsVIP ? 1 : 0)
      .set("BirthPlace", user.BirthPlace)
      .set("BirthDay", user.BirthDay)
      .set("Phone1", user.Phone1)
      .set("Phone2", user.Phone2)
      .set("Address", user.Address)
      .set("City", user.City)
      .set("Province", user.Province)
      .set("State", user.State)
      .set("PhotoDoc", user.PhotoDoc)
      .set("PhotoGuest", user.PhotoGuest)
      .set("CreateAt", new_moment.format("YYYY-MM-DD"))
      .set("UpdateAt", new_moment.format("YYYY-MM-DD"));
    var result = executeScalar(qry);

    return Number(result);
  }
}
